src/EditorFeatures/Core/Implementation/AutomaticCompletion/Sessions/EditorBraceCompletionSession.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using Microsoft.CodeAnalysis.BraceCompletion;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Text.BraceCompletion;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.AutomaticCompletion.Sessions
+{
+    internal class EditorBraceCompletionSession : IEditorBraceCompletionSession
+    {
+        private readonly IBraceCompletionService _braceCompletionService;
+
+        public EditorBraceCompletionSession(IBraceCompletionService braceCompletionService)
+        {
+            _braceCompletionService = braceCompletionService;
+        }
+
+        public BraceCompletionResult? GetChangesAfterReturn(IBraceCompletionSession session, CancellationToken cancellationToken)
+        {
+            var context = GetBraceCompletionContextFromSession(session);
+            if (context == null)
+            {
+                return null;
+            }
+
+            return _braceCompletionService.GetTextChangeAfterReturnAsync(context, supportsVirtualSpace: true, cancellationToken).WaitAndGetResult(cancellationToken);
+        }
+
+        public virtual bool AllowOverType(IBraceCompletionSession session, CancellationToken cancellationToken)
```

Comment:
don't love that we have virtual behavior in a concrete type.  will have to see why we do this and if we should change.